dts/arm/st/u3/stm32u3.dtsi

```diff
@@ -5,6 +5,8 @@
  */
 
 #include <arm/armv8-m.dtsi>
+#include <zephyr/dt-bindings/adc/adc.h>
+#include <zephyr/dt-bindings/adc/stm32u3_adc.h>
```

Comment:
You haven't created stm32u3_adc.h file. Besides, you should reuse stm32u5_adc.h here anyway.